### PR TITLE
Added nutrimouse dataset and example

### PR DESCRIPTION
**Pull request by rflperry**
Added nutrimouse multiview genomics dataset and example using a variety of methods to address feedback we received.

Still want to add as part of this PR a getting started section that links or includes this.

### CODE REVIEW

File: examples/datasets/plot_nutrimouse.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+demonstrate applications to the Nutrimouse dataset from a nutrition study on
+mice. The data measures 40 mice and has two views: expression levels of
+potentially relevant genes and concentrations of certain fatty acids. Each
+mouse has two labels: it's genetic type and diet.
+
+[1] P. Martin, H. Guillou, F. Lasserre, S. Déjean, A. Lan, J-M.
+    Pascussi, M. San Cristobal, P. Legrand, P. Besse, T. Pineau.
+    "Novel aspects of PPARalpha-mediated regulation of lipid and
+    xenobiotic metabolism revealed through a nutrigenomic study."
+    Hepatology, 2007.
```
Comment (agramfort) [inline examples/datasets/plot_nutrimouse.py:16]: you have a formatting issue:

https://297-206838300-gh.circle-artifacts.com/0/dev/auto_examples/datasets/plot_nutrimouse.html#sphx-glr-auto-examples-datasets-plot-nutrimouse-py

File: examples/datasets/plot_nutrimouse.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+diet_names = dataset['diet_names']
+genotype_names = dataset['genotype_names']
+plt.figure(figsize=(5, 5))
+for genotype_idx, cmap in enumerate((cm.Blues, cm.Reds)):
+    for diet_idx in range(5):
+        X_idx = np.where((y == (genotype_idx, diet_idx)).all(axis=1))
+        color = cmap((diet_idx + 1) / 6)
+        label = diet_names[diet_idx] + f' ({genotype_names[genotype_idx]})'
+        plt.scatter(*zip(*X_mvmds[X_idx]), color=color, label=label)
```
Comment (agramfort) [inline examples/datasets/plot_nutrimouse.py:60]: you should use both color and a marker. Also you use a continuous color scale for a categorical (unordered information). Can you try to improve?

File: examples/datasets/plot_nutrimouse.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+sca_kwargs = {'alpha': 0.7, 's': 10}
+
+f, axes = plt.subplots(1, 3, figsize=(10, 3))
+axes[0].scatter(*zip(*X_mvmds), c=Xs_labels, **sca_kwargs)
+axes[0].set_title('Multiview Kmeans Clusters')
+axes[1].scatter(*zip(*X_mvmds), c=X1_labels, **sca_kwargs)
+axes[1].set_title('View 1 Kmeans Clusters')
+axes[2].scatter(*zip(*X_mvmds), c=X2_labels, **sca_kwargs)
+axes[2].set_title('View 2 Kmeans Clusters')
```
Comment (agramfort) [inline examples/datasets/plot_nutrimouse.py:91]: the yellow dots in the scaller are not really visible.

File: examples/datasets/plot_nutrimouse.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+axes[0].set_ylabel('MVMDS component 2')
+axes[0].set_title('Multiview Kmeans Clusters')
+plt.show()
+
+###############################################################################
+# Decomposition using AJIVE
+# -------------------------
+#
+# We can also apply joint decomposition tools to find features across views
+# that are jointly related. Using AJIVE, we can find genes and lipids that are
```
Comment (agramfort) [inline examples/datasets/plot_nutrimouse.py:106]:
```suggestion
# that are jointly related. Using :class:`mvlearn.AJIVE`, we can find genes and lipids that are
```
something like this should work. It allows to have cross-links and point to the classes and more
details as AJIVE is certainly mysterious for most readers.

File: examples/datasets/plot_nutrimouse.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+print(f'Canonical correlations: {cca.canon_corrs(Xs_cca)}')
+f, axes = plt.subplots(1, 2, figsize=(12, 2))
+gene_ticks = [n if i in [31, 57, 76, 88] else '' for i,
+              n in enumerate(dataset['gene_feature_names'])]
+g = sns.heatmap(cca.loadings_[0].T,
+                yticklabels=['Component 1', 'Component 2'],
+                cmap="RdBu_r", ax=axes[0],
+                xticklabels=gene_ticks)
+g.set_xticklabels(gene_ticks, rotation=90)
+axes[0].set_title('Gene expression loadings')
```
Comment (agramfort) [inline examples/datasets/plot_nutrimouse.py:164]: putting heatmaps in subplots makes the fonts too small I think. It's not really readable in:

https://297-206838300-gh.circle-artifacts.com/0/dev/auto_examples/datasets/plot_nutrimouse.html#inference-using-regularized-cca

File: tests/datasets/test_nutrimouse.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
+from mvlearn.datasets import load_nutrimouse
+import numpy as np
+import numbers
```
Comment (agramfort) [inline tests/datasets/test_nutrimouse.py:3]:
```suggestion
import numbers
import numpy as np
from mvlearn.datasets import load_nutrimouse
```

**Review by agramfort — APPROVED** [review]
feel free to MRG !

thx @rflperry

**Comment by rflperry** [comment]
@gavinmischler To address feedback I added some motivation and small examples to the main page, and linked to the CCA and UCI clustering examples which got moved out of the paper. I also reformatted your UCI example to better reflect the paper. If that looks good rendered I'll go ahead and merge.

**Comment by gavinmischler** [comment]
> @gavinmischler To address feedback I added some motivation and small examples to the main page, and linked to the CCA and UCI clustering examples which got moved out of the paper. I also reformatted your UCI example to better reflect the paper. If that looks good rendered I'll go ahead and merge.

In the last figure, the bottom left appears to be the PCA embedding even though it should be MVMDS, or am I looking at it wrong?

File: docs/overview.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        from mvlearn.cluster import MultiviewKMeans
+        # X1 and X2 are data matrices, each with n samples
+        Xs = [X1, X2] # multiview data
+        labels = MultiviewKMeans().fit_predict(Xs)
+
+
+Highlighted full examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- `Nutrimouse datasetcase study <auto_examples/datasets/plot_nutrimouse.html>`_:
```
Comment (gavinmischler) [inline docs/overview.rst:63]:
```suggestion
- `Nutrimouse dataset case study <auto_examples/datasets/plot_nutrimouse.html>`_:
```

**Comment by rflperry** [comment]
> @rflperry In the last figure, the bottom left appears to be the PCA embedding even though it should be MVMDS, or am I looking at it wrong?
@gavinmischler should be fixed in this build.

**Pull request merged**